src/skills/doc-testing/scripts/test-skill.sh

```
@@ -0,0 +1,818 @@
+            # --permission-mode bypassPermissions allows file reads without prompts
+            # --max-turns limits iterations to prevent runaway
+            if (cd "$SKILL_DIR" && claude -p "$prompt" \
+                --output-format json \
+                --max-turns 5 \
+                --permission-mode bypassPermissions \
+                > "$output_file" 2>&1); then
+                cat "$output_file"
+            else
+                # Capture both the output and error
+                local exit_code=$?
+                local output_content=$(cat "$output_file" 2>/dev/null | tr '\n' ' ' | head -c 1000 | sed 's/"/\\"/g')
```

Shellcheck (0.11.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)